src/_posts/2019-12-30-backstopjs-tutorial.md

```diff
@@ -0,0 +1,257 @@
+
+This article will explain how to set up BackstopJS and some good practices I've picked up from using it in anger. You'll need to have some awareness of Docker Compose and Yarn or NPM.
+
+![An image of BackstopJS's interactive diff between two snapshots of a website. A red line separates each snapshot, demonstrating that the difference between them is how they render an image on the page; one is blurry and the other is not]({{site.baseurl}}/assets/custom/img/blog/2019-12-30-backstopjs/backstopjs-scrubber.png)
+
+## Why
+
+Why use visual regression testing? I'll assume that you appreciate why testing as a general practice is necessary, so here are several scenarios that visual regression tests will catch and other testing techniuqes won't:
+
+* CSS regressions: Often we'll make a change to a CSS style to move that one button into the space we'd like it to be, but how do we know that it hasn't had an undesirable knock-on effect on some other part of the website that uses the same CSS class? Instead of checking every page manually (which we will most often forget to do), run your visual regression tests.
+* Responsiveness: Most often we're working on a 1080p screen, but many (perhaps most) of our users will be using their smartphones. Visual regression tests can test an assortment of different screen sizes, giving you confidence that your changes haven't broken the responsive page at other sizes and saving you time manually testing the page at different resolutions.
+* Dependabot: It's a wonderful tool that saves you from manually keeping all of your dependencies up-to-date. In my current team [we use dependabot aggressively](/2019/02/24/taming-dependabot/); we have extensive testing and auto-merge any Dependabot PRs that pass all our tests. In fact, in terms of PRs or commits it's the most active member of our team by far. However, you can't place your trust in Dependabot's auto-merge in the front-end if you aren't testing for visual regressions. Before we introduced visual regression testing we had instances where Dependabot would automatically update to a new version of a dependency that introduced a visual change (at one point even removing a prominent navigation link), and it would be automatically merged and deployed to production. We trust Dependabot to change our software's behaviour because we have extensive tests of that behaviour, and we trust it to change our software's visual appearance because we have visual regression tests.
```

Comment:
Dependabot is a tool not a technique.

Reply:
Dependabot is being described as a "scenario" here, the "technique" is visual regression testing.